File: src/SelectionSort.py
# Implementation of Selection sort algorithm for sorting
class SelectionSort:

    # Sort input array
    def sort(A):

        # Swap array elements at indices u and v
        def swap(A, u, v):
            temp = A[u]
            A[u] = A[v]
            A[v] = temp

        # Get length of array
        a_len = len(A)

        # Iterate over the array
        for i in range(0, a_len - 1):
            # Set the current minimum
            curr_min = i

            # Scan the rest of array
            for j in range(i+1, a_len):
                
                # If the element A[j] is less that the
                # current minimum then set the new minimum.
                if(A[j] < A[curr_min]):
                   curr_min = j

            # Swap A[i] with the current minimum.
            swap(A, i, curr_min)

        # Return the sorted array
        return A

        

File: java-src/SelectionSort.java
// Selection sort algorithm
public class SelectionSort
{
    private int[] A;

    // Constructor
    public SelectionSort(int[] data)
    {
        A = data;
    }

    public int[] sort()
    {
        int a_len = A.length;

        // Iterate over the array
        for(int i=0; i<a_len-1; i++)
        {
            // Set the current minimum
            int curr_min = i;
            
            // Scan the rest of the array
            for(int j=i+1; j<a_len; j++)
            {
                // If the element A[j] is less than the current minimum 
                // then set the new minimum.
                if(A[j] < A[curr_min])
                {
                    curr_min = j;
                }
            }

            // Swap A[i] with the current minimum.
            swap(A, i, curr_min);
        }

        // Return the sorted array
        return A;
    }

    // Swap array elements at A[u] and A[v].
    private void swap(int[] A, int u, int v)
    {
        int temp;
        
        temp = A[u];
        A[u] = A[v];
        A[v] = temp;
    }
}